lib/std/http/method.zig

```diff
@@ -14,4 +14,87 @@ pub const Method = enum {
     OPTIONS,
     TRACE,
     PATCH,
+
+    /// Returns true if a request of this method is allowed to have a body
+    /// Returns null if it also depends on other conditions
+    pub fn requestHasBody(self: Method) ?bool {
+        return switch (self) {
+            .GET => false,
+            .HEAD => false,
+            .POST => true,
+            .PUT => true,
+            .DELETE => null,
+            .CONNECT => false,
+            .OPTIONS => false,
+            .TRACE => false,
```

Comment:
I think `TRACE` can have a body too?

Reply:
gonna rework the doc comment to reflect this